gapic/schema/wrappers.py

```diff
@@ -866,13 +866,19 @@ def paged_result_field(self) -> Optional[Field]:
         """Return the response pagination field if the method is paginated."""
         # If the request field lacks any of the expected pagination fields,
         # then the method is not paginated.
-        for page_field in ((self.input, int, 'page_size'),
-                           (self.input, str, 'page_token'),
+        for page_field_token in ((self.input, str, 'page_token'),
                            (self.output, str, 'next_page_token')):
-            field = page_field[0].fields.get(page_field[2], None)
-            if not field or field.type != page_field[1]:
+            field = page_field_token[0].fields.get(page_field_token[2], None)
+            if not field or field.type != page_field_token[1]:
                 return None
 
+        page_fields = (self.input.fields.get('max_results', None),
+                       self.input.fields.get('page_size', None))
+        page_field_size = next(
+            (field for field in page_fields if field is not None), None)
```

Comment:
Nit: we don't have to explicitly check for `None`.
```python
next((field for field in page_fields if field), None)
```

Reply:
I just wanted to be extra explicit in case there was some false-ish value `field` might have. I suppose that's not the case for the `Field` class though.